src/main/java/stream/flarebot/flarebot/util/general/GeneralUtils.java

```diff
@@ -137,28 +133,242 @@ public static String getProgressBar(Track track) {
                 " " + GeneralUtils.percentageFormat.format(percentage) + "%";
     }
 
-    private static char getPrefix(Guild guild) {
-        return guild == null ? Constants.COMMAND_CHAR : FlareBotManager.instance().getGuild(guild.getId()).getPrefix();
+    /**
+     * Gets a String stacktrace from a {@link Throwable}
+     *
+     * @param e the {@link Throwable}.
+     * @return A string stacktrace
+     */
+    public static String getStackTrace(Throwable e) {
+        StringWriter writer = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(writer);
+        e.printStackTrace(printWriter);
+        printWriter.close();
+        return writer.toString();
     }
 
-    public static String formatCommandPrefix(Guild guild, String usage) {
-        String prefix = String.valueOf(getPrefix(guild));
-        if (usage.contains("{%}"))
-            return usage.replaceAll("\\{%}", prefix);
-        return usage;
+    /**
+     * Gets an int from a String
+     *
+     * @param s The string to get an int from.
+     * @param defaultValue The default int value to get in case conversion fails
+     * @return an int
+     */
+    public static int getInt(String s, int defaultValue) {
+        try {
+            return Integer.parseInt(s);
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
+    /**
+     * Gets a long from a String
+     *
+     * @param s The string to get a long from.
+     * @param defaultValue The default long value to get in case conversion fails
+     * @return an int
+     */
+    public static long getLong(String s, long defaultValue) {
+        try {
+            return Long.parseLong(s);
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Get a Joda Period from the input string. This will convert something like `1d20s` to 1 day and 20 seconds in the
```

Comment:
Oh so you do know how to write docs? :bloblul: Just teasing you but make the other docs more like this!

Reply:
Also make sure to link where possible